packages/neuron-wallet/src/database/chain/entities/hd-public-key-info.ts

```diff
@@ -0,0 +1,70 @@
+import { Entity, Column, PrimaryGeneratedColumn, Index, CreateDateColumn } from 'typeorm'
+import HdPublicKeyInfoModel from 'models/keys/hd-public-key-info'
+import { AddressType } from 'models/keys/address'
+
+@Entity()
+export default class HdPublicKeyInfo {
+
+  @PrimaryGeneratedColumn()
+  id!: number
+
+  @Column({
+    type: "varchar",
+  })
+  @Index()
+  walletId!: string
+
+  @Column({
+    type: 'varchar',
+  })
+  path!: string
+
+  @Column({
+    type: 'varchar',
+  })
+  @Index()
+  address!: string
+
+  @Column()
+  addressType!: AddressType
+
+  @Column()
+  @Index()
+  addressIndex!: number
+
+  @Column({
+    type: 'varchar',
+  })
+  publicKeyInBlake160!: string
+
+  @Column({
+    type: 'varchar',
+    default: null
+  })
+  description?: string
+
+  @CreateDateColumn({
+    type: "varchar",
+    default: () => "CURRENT_TIMESTAMP"
+  })
+  createdAt!: Date;
+
+  public static fromModel(model: HdPublicKeyInfoModel): HdPublicKeyInfo {
+    const publicKeyInfo = new HdPublicKeyInfo()
+
+    publicKeyInfo.walletId = model.walletId
+    publicKeyInfo.path = model.path
+    publicKeyInfo.address = model.address
+    publicKeyInfo.addressType = model.addressType
+    publicKeyInfo.addressIndex = model.addressIndex
+    publicKeyInfo.publicKeyInBlake160 = model.publicKeyInBlake160
+    publicKeyInfo.description = model.description
+
+    return publicKeyInfo
+  }
+
+  public static fromObject(...args: any[]): HdPublicKeyInfo {
+    const model = HdPublicKeyInfoModel.fromObject.apply(HdPublicKeyInfoModel, args)
```

Comment:
It is just to proxy anything from the arguments to the other method. Feel free to suggest if any other better ways to way to do this.

Reply:
`HdPublicKeyInfoModel.fromObject(...args)` could be enough.

Reply:
Then it can't get around the checks by TS.
![Screen Shot 2020-08-27 at 12 13 28 PM](https://user-images.githubusercontent.com/1587352/91383458-c3eed300-e85e-11ea-8bc9-9adf2c385ddd.png)

Reply:
Declare the `args` as `Parameter<typeof HdPublicKeyInfoModel.fromObject>`

Reply:
It works. Thanks a lot for this TS technique.